python/pyarrow/tests/test_schema.py

```diff
@@ -87,3 +87,10 @@ def test_schema_equals(self):
         del fields[-1]
         sch3 = A.schema(fields)
         assert not sch1.equals(sch3)
+
+
+class TestField(unittest.TestCase):
+    def test_empty_field(self):
+        f = arrow.Field()
+        # This used to segfault on empty field
+        repr(f)
```

Comment:
In some other classes we have a convenience function for this: https://github.com/apache/arrow/blob/master/python/pyarrow/table.pyx#L63